torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3031,6 +3031,11 @@ bool check_same_size(const std::vector<at::Tensor>& input_tensors) {
   return true;
 }
 
+at::cuda::CUDAEvent* safeGetCudaEventPtr(
+    const std::shared_ptr<at::cuda::CUDAEvent>& event) {
+  return event ? event.get() : nullptr;
```

Comment:
is event.get() returning something other than nullptr in the case where bool(event) is true?